dex-it/src/test/scala/com/wavesplatform/it/sync/MarketOrderSuite.scala

```diff
@@ -0,0 +1,380 @@
+      val marketPrice       = 0.5.usd
+      val marketOrderAmount = 150.waves
+      val ordersAmount      = 120.waves
+
+      placeOrders(alice, wavesUsdPair, OrderType.SELL)(
+        30.waves -> 0.3.usd,
+        40.waves -> 0.4.usd,
+        50.waves -> 0.5.usd,
+      )
+
+      node.waitOrderProcessed(
+        wavesUsdPair,
+        node.placeMarketOrder(node.prepareOrder(bob, wavesUsdPair, OrderType.BUY, marketOrderAmount, marketPrice, fee)).message.id)
+
+      val orderBook = node.orderBook(wavesUsdPair)
+      orderBook.bids should be(empty)
+      orderBook.asks should be(empty)
+
+      /* market order fee value depends on matched orders in proportion */
+      node.accountBalances(bob.toAddress.toString)._1 should be(bobWBefore + ordersAmount - fee * ordersAmount / marketOrderAmount)
+
+      node.assetBalance(bob.toAddress.toString, UsdId.toString).balance should be(
+        bobUBefore - 0.3.usd * 30.waves / 1.waves - 0.4.usd * 40.waves / 1.waves - 0.5.usd * 50.waves / 1.waves)
+    }
+
+    "should be removed from order book when the restriction by tokens count has been reached (SELL)" in {
+      // val bobWBefore        = node.accountBalances(bob.toAddress.toString)._1
+      val bobUBefore        = node.assetBalance(bob.toAddress.toString, UsdId.toString).balance
+      val marketPrice       = 0.1.usd
+      val marketOrderAmount = 72.waves
+      // val ordersAmount      = 36.waves
+
+      placeOrders(alice, wavesUsdPair, OrderType.BUY)(
+        10.waves -> 0.2.usd,
+        14.waves -> 0.3.usd,
+        12.waves -> 0.4.usd,
+      )
+
+      node.waitOrderProcessed(
+        wavesUsdPair,
+        node.placeMarketOrder(node.prepareOrder(bob, wavesUsdPair, OrderType.SELL, marketOrderAmount, marketPrice, fee)).message.id)
+
+      val orderBook = node.orderBook(wavesUsdPair)
+      orderBook.bids should be(empty)
+      orderBook.asks should be(empty)
+
+      /* market order fee value depends on matched orders in proportion
+       *
+       *  TODO: bug DEX-454
+       *  node.accountBalances(bob.toAddress.toString)._1 should be(bobWBefore - ordersAmount - fee * ordersAmount / marketOrderAmount)
+       */
+
+      node.assetBalance(bob.toAddress.toString, UsdId.toString).balance should be(
+        bobUBefore + 0.2.usd * 10.waves / 1.waves + 0.3.usd * 14.waves / 1.waves + 0.4.usd * 12.waves / 1.waves)
+
+    }
+
+    "should be removed from order book when there are no suitable orders by limit of the price (BUY)" in {
+      val accountBalanceWBefore = 1.waves
+      val marketOrderAmount     = 200.waves
+      val marketPrice           = 0.1.usd
+      val anotherOrderAmount    = 1.waves
+
+      val account = createAccountWithBalance()(
+        1.waves -> None,
+        100.usd -> Some(UsdId.toString)
+      )
+
+      val creationTime = System.currentTimeMillis();
+
+      node.placeOrder(node.prepareOrder(alice, wavesUsdPair, OrderType.SELL, 200.waves, marketPrice, fee)).message.id
```

Comment:
You don't need `.message.id` if you won't await order status

So, if you want to place an order and don't care about its status, just use 
`node.placeOrder(node.prepareOrder(alice, wavesUsdPair, OrderType.SELL, 200.waves, marketPrice, fee))`

Please, check this issue all over the test

Reply:
ok